bip-tap.mediawiki

```diff
@@ -398,18 +398,17 @@ derived using the first input outpoint as a sort of chain randomness to ensure
 uniqueness within the system as well as the output index and asset type to
 prevent collisions.
 The <code>asset_group_key</code> is derived as follows:
-* <code>asset_group_key</code> = <code>asset_key_internal +
-sha256(asset_key_internal || genesis_outpoint || output_index || asset_type)
-</code>
+* <code>asset_group_key = TapTweak(asset_internal_group_key, asset_group_script_root)</code>
 
 where:
-* <code>asset_key_internal</code> is a 32-byte public key as defined by
-[[../master/bip-0340.mediawiki|BIP-340]]
-* <code>genesis_outpoint</code> is the first previous input outpoint used in the
-asset genesis transaction serialized in Bitcoin wire format.
-* <code>output_index</code> (4 byte, big-endian) is the index  of the output
-which contains the unique Taproot Asset commitment in the genesis transaction. 
-* <code>asset_type</code> (1 byte) is the type of the asset being minted.
+* <code>asset_internal_group_key = SingleTweak(asset_raw_key, asset_id)</code> 
```

Comment:
So just to clarify: this is just an _initial binding_ to the assetID that created the group key right? So then for future minting events, all is needed is that a valid witness over the virtual transaction is presented, and there's no new layer of tweaking needed to arrive at the key things _should_ be validated against?

Reply:
Yes, this binds the `asset_id` of the original minting transaction (the first tranche) to the group key. This tweak won't later change.

Reply:
Can't a 3rd party that is aware of the the final tweaked key and a following reissuance reuse a script witness from a previous mint?

Before we used the final tweaked key to produce a new signature, and the witness (signature) also committed to the new asset being reissued - AFAICT now we are able to provide a witness that doesn't commit to the new assetID, and can be valid without owning the private key for the final tweaked key.

What am I missing here?

Reply:
I think maybe the discussion here answers your question? https://github.com/Roasbeef/bips/pull/33#issuecomment-1600614525

Reply:
IIUC that comment describes the flow for reissuance via the 'key path' of the group key. What I'm thinking about is the 'script path'.

In that case, if (and only if) the script itself does not require a signature, then we would not be signing over the virtual TX, and the script could be replayed.

I think this is addressed by just not having scripts that do not also require signatures, which is similar to existing best practices for on-chain scripting. IMO this would be worth noting somewhere since the consequences are different here compared to on-chain UTXOs or even tap assets (where you would also need to control the anchor output to take advantage of script replay).

Reply:
That is true, good catch!

But yeah, I think this will be similar to a normal bitcoin script with no signature. The difference is that we here expect to be re-using the script tree as many time as we like, while a Bitcoin UTXO can be spent only once.

I don't know if there are any real solutions to it though. Even if we had the witness commit to the `asset_id`, the script path witness could be re-used/re-created/replayed for another mint just by changing the asset_id (since no private key is required, anyone could do this). 

Am I understanding it correctly now?

(and yeah, worth mentioning somewhere in the docs)